### PR TITLE
Added test case for PUBDEV-4026.

### PR DESCRIPTION
**Pull request by vpatryshev**
Passes.

### CODE REVIEW

**Comment by vpatryshev** [comment]
@h2o-ops please test!

File: h2o-core/src/main/java/water/parser/Parser.java [hunk 1]
```diff
@@ -231,7 +231,7 @@ ParseWriter streamParseZip( final InputStream is, final StreamParseWriter dout,
    *  Implements a classic double-buffer reader.
    */
   final static class StreamData implements ParseReader {
-    public static int bufSz = 64*1024;
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/parser/Parser.java]: Why was it increased?

File: h2o-algos/src/main/java/hex/glm/GLM.java [hunk 2]
```diff
@@ -1117,7 +1117,7 @@ public void computeImpl() {
         }
         _job.update(_workPerIteration,"iter=" + _state._iter + " lmb=" + lambdaFormatter.format(_state.lambda()) + "deviance trn/tst= " + devFormatter.format(trainDev) + "/" + devFormatter.format(testDev) + " P=" + ArrayUtils.countNonzeros(_state.beta()));
       }
-      if(_state._iter >= _parms._max_iterations)
+      if(_state._iter >= _parms._max_iterations && _parms._max_iterations > 1)
```
Comment (mmalohlava) [inline h2o-algos/src/main/java/hex/glm/GLM.java]: I believe, this is unrelated to SVMLight parser.

File: h2o-core/src/test/java/water/parser/ParseFolderTestBig.java [hunk 3]
```diff
@@ -50,10 +52,38 @@ public void testKDDCup() {
...
+      ParseWriter.ParseErr[] errors = globalSetup._errs;
+      
+      assertTrue("Got errors: " + Arrays.toString(errors), errors.length == 0);
+      Job<Frame> job = ParseDataset.parse(Key.make("PUBDEV4026.hex"), new Key[]{nfs._key}, true, globalSetup, false)._job;
+      while (job.progress() < 1.0) {
+        System.out.print(((int) (job.progress() * 1000.0)) / 10.0 + "% ");
+        try {
+          Thread.sleep(1000);
+        } catch (InterruptedException ignore) { /*comment to disable ideaJ warning*/}
+      }
```
Comment (mmalohlava) [inline h2o-core/src/test/java/water/parser/ParseFolderTestBig.java]: I do not understand this  part. There is no check for results.

Comment (michalkurka) [inline h2o-core/src/test/java/water/parser/ParseFolderTestBig.java]: It checks that there are no parser errors (line 65):

    assertTrue("Got errors: " + Arrays.toString(errors), errors.length == 0);

Comment (mmalohlava) [inline h2o-core/src/test/java/water/parser/ParseFolderTestBig.java]: Yup, but i am missing something:
  - if code checks just guesser then it does not need to parse 
  - if code checks that there are no guesser errors and no parser errors, then code is missing one assert after parse.

Comment (michalkurka) [inline h2o-core/src/test/java/water/parser/ParseFolderTestBig.java]: Yep, you are right - I was too quick with sending the comment (updated it in the meantime)

**Comment by michalkurka** [comment]
@vpatryshev okay, got it, this doesn't fix the issue it makes it work for this dataset and datasets of similar characteristics

My opinion is that we should not merge this PR because it doesn't resolve the problem to the full extent.

@mmalohlava what is your opinion?

**Comment by mmalohlava** [comment]
@michalkurka agree, that just hides a problem. We need to look deeper and fix cause for it.

**Comment by vpatryshev** [comment]
@h2o-ops please test!

**Comment by vpatryshev** [comment]
So, either this goes in as a solution to the jira case, or I'll have to investigate a couple of related problems that are causing all this, or I'll have to work on an algorithm.

It's more of a business decision. Michal?

**Comment by mmalohlava** [comment]
:) it is not business decision (honestly i do not know what it means in this context) - we just need a clean patch, which solves the issue, not hide it.

**Comment by vpatryshev** [comment]
By business decision I mean the choice whether we release fast (customers
waiting) or achieve perfection (I personally prefer the latter, in an ideal
world).

Thanks,
-Vlad
